File: conversation-websocket-tools/node/create-agent.ts
import { PhonicClient } from "phonic";
import { phonicApiKey } from "./env-vars";

const client = new PhonicClient({
  apiKey: phonicApiKey,
});

async function createAgent() {
  await client.agents.create({
    name: "agent-websocket-tools",
    welcome_message: "Hi there. How can I help you today?",
    system_prompt: "Be helpful, friendly, and concise.",
    audio_format: "mulaw_8000",
  });
}

createAgent();

File: conversation-websocket-tools/python/create_agent.py
import os

from dotenv import load_dotenv
from phonic import Phonic

load_dotenv(".env.local")

client = Phonic(
    api_key=os.getenv("PHONIC_API_KEY"),
)

system_prompt = """
Be helpful, friendly, and concise.
You have 1 tool, called find_flights_sync.
If you are about to call a tool, you can say something like "Just wait a moment while I look up flights".
The result of the tool call may not sound natural if you read it directly,
so please convert the tool call response to something more natural when saying it.
For example, don't read out entire bulleted lists.
"""

client.agents.create(
    name="agent-websocket-find-flights",
    welcome_message="Hi there. What flights are you interested in?",
    system_prompt=system_prompt,
    audio_format="mulaw_8000",
    tools=["find_flights_sync"],
)
